runtime/oti/j9accessbarrier.h

```diff
@@ -634,8 +634,11 @@ typedef struct J9IndexableObject* mm_j9array_t;
 				: J9OBJECT_UDATA_STORE_VM(javaVM, object, TMP_OFFSETOF_J9OBJECT_CLAZZ, (UDATA)(((UDATA)(value) & ~((UDATA)(J9_REQUIRED_CLASS_ALIGNMENT - 1))) | J9OBJECT_FLAGS_FROM_CLAZZ_VM((javaVM), (object)))))
 
 #define J9OBJECT_MONITOR_OFFSET(vmThread,object) (J9OBJECT_CLAZZ(vmThread, object)->lockOffset)
+#define J9OBJECT_MONITOR_OFFSET_VM(vm,object) (J9OBJECT_CLAZZ_VM(vm, object)->lockOffset)
```

Comment:
I suggest pairs of macros like these should only have one real definition:
```
#define J9OBJECT_MONITOR_OFFSET(vmThread, object) J9OBJECT_MONITOR_OFFSET_VM(J9VMTHREAD_JAVAVM(vmThread), object)
#define J9OBJECT_MONITOR_OFFSET_VM(vm, object) (J9OBJECT_CLAZZ_VM(vm, object)->lockOffset)
```

Reply:
That's extra instructions for no reason, given these are one-liners.

Reply:
What extra instructions? The result should be the same unless one of the macros of a pair is broken.

Reply:
thread->VM->vmCompressedFlag is (trivially) longer than thread->threadCompressedFlag. It will make no difference in the normal builds, since neither anchor is used.

Reply:
When there are differences like that example, I agree. Do all the macros you're adding have that property?

Reply:
There's one multi-line addition, but it mirrors all the other macros in this file (they do not forward to each other).